pkg/handler/iam_handler_test.go

```diff
@@ -1102,6 +1102,507 @@ func TestDo(t *testing.T) {
+							"user:test-folder-user@example.com",
+						},
+						Role: "roles/accessapproval.approver",
+					},
+				},
+			},
+			wantProjectsPolicy: &iampb.Policy{},
+		},
+		{
+			name: "failed_with_orgs_server_get_iam_policy_error",
+			organizationsServer: &fakeServer{
+				policy:          &iampb.Policy{},
+				getIAMPolicyErr: fmt.Errorf("Get IAM policy encountered error: Internal Server Error"),
+			},
+			foldersServer: &fakeServer{
+				policy: &iampb.Policy{},
+			},
+			projectsServer: &fakeServer{
+				policy: &iampb.Policy{},
+			},
+			request: &v1alpha1.IAMRequest{
+				ResourcePolicies: []*v1alpha1.ResourcePolicy{
+					{
+						Resource: "organizations/foo",
+						Bindings: []*v1alpha1.Binding{
+							{
+								Members: []string{
+									"user:test-org-userA@example.com",
+									"user:test-org-userB@example.com",
+								},
+								Role: "roles/accessapproval.approver",
+							},
+						},
+					},
+					{
+						Resource: "projects/baz",
+						Bindings: []*v1alpha1.Binding{
+							{
+								Members: []string{
+									"user:test-project-user@example.com",
+								},
+								Role: "roles/bigquery.dataViewer",
+							},
+						},
+					},
+				},
+			},
+			wantPolicies: []*v1alpha1.IAMResponse{
+				{
+					Resource: "projects/baz",
+					Policy:   &iampb.Policy{},
+				},
+			},
+			wantErrSubstr:           "Get IAM policy encountered error: Internal Server Error",
+			wantOrganizationsPolicy: &iampb.Policy{},
+			wantFoldersPolicy:       &iampb.Policy{},
+			wantProjectsPolicy: &iampb.Policy{
+				Bindings: []*iampb.Binding{},
+			},
+		},
+		{
+			name: "failed_with_projects_server_set_iam_policy_error",
+			organizationsServer: &fakeServer{
+				policy: &iampb.Policy{},
+			},
+			foldersServer: &fakeServer{
+				policy: &iampb.Policy{},
+			},
+			projectsServer: &fakeServer{
+				policy:          &iampb.Policy{},
+				setIAMPolicyErr: fmt.Errorf("Set IAM policy encountered error: Internal Server Error"),
+			},
+			request: &v1alpha1.IAMRequest{
+				ResourcePolicies: []*v1alpha1.ResourcePolicy{
+					{
+						Resource: "folders/bar",
+						Bindings: []*v1alpha1.Binding{
+							{
+								Members: []string{
+									"user:test-folder-user@example.com",
+								},
+								Role: "roles/cloudkms.cryptoOperator",
+							},
+						},
+					},
+					{
+						Resource: "projects/baz",
+						Bindings: []*v1alpha1.Binding{
+							{
+								Members: []string{
+									"user:test-project-user@example.com",
+								},
+								Role: "roles/bigquery.dataViewer",
+							},
+						},
+					},
+				},
+			},
+			wantPolicies: []*v1alpha1.IAMResponse{
+				{
+					Resource: "folders/bar",
+					Policy:   &iampb.Policy{},
+				},
+			},
+			wantErrSubstr:           "Set IAM policy encountered error: Internal Server Error",
+			wantOrganizationsPolicy: &iampb.Policy{},
+			wantFoldersPolicy: &iampb.Policy{
+				Bindings: []*iampb.Binding{},
+			},
+			wantProjectsPolicy: &iampb.Policy{},
+		},
+	}
```

Comment:
How about adding a case for mixing multiple bindings together? Expired, not expired, requester bindings, non-requester bindings, etc.

Reply:
Added.